### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by XiongKezhi**
Do we also need to include warnings checks and coverage checks in this plugin's doc?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,12 +3,41 @@
...
 [![Jenkins](https://ci.jenkins.io/job/Plugins/job/github-checks-plugin/job/master/badge/icon?subject=Jenkins%20CI)](https://ci.jenkins.io/job/Plugins/job/github-checks-plugin/job/master/)
 [![GitHub Actions](https://github.com/jenkinsci/github-checks-plugin/workflows/CI/badge.svg?branch=master)](https://github.com/jenkinsci/github-checks-plugin/actions)
+[![Codacy Badge](https://app.codacy.com/project/badge/Grade/2c7fa67496a743778ca60cc9604212d2)](https://www.codacy.com/gh/jenkinsci/github-checks-plugin?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=jenkinsci/github-checks-plugin&amp;utm_campaign=Badge_Grade)
+[![codecov](https://codecov.io/gh/jenkinsci/github-checks-plugin/branch/master/graph/badge.svg)](https://codecov.io/gh/jenkinsci/github-checks-plugin)
 
+![GitHub Checks Plugin Cover](docs/images/github-checks-plugin-cover.png)
 
-This a [Google Summer of Code](https://summerofcode.withgoogle.com/) project for [Jenkins Organization](https://www.jenkins.io/), it is still under development.
+This plugin helps Jenkins plugins publish checks to GitHub through [GitHub Checks API](https://docs.github.com/en/rest/reference/checks#runs).
```
Comment (timja) [inline README.md:11]: This plugin is repeated in this paragraph and the next...

```suggestion
Publishes checks to GitHub through [GitHub Checks API](https://docs.github.com/en/rest/reference/checks#runs).
```

File: README.md [hunk 2]
```diff
@@ -3,12 +3,41 @@
...
 
-## Useful links:
-* [Project Idea](https://www.jenkins.io/projects/gsoc/2020/project-ideas/github-checks/)
-* [Project Design](https://docs.google.com/document/d/1hVQTd9jKw0sx8JQR8KjbM-7lWW84e2vFjmmkzBpBbSk/edit?usp=sharing)
-* [Meeting Notes](https://docs.google.com/document/d/1TZLmu3nBPbwUjzLVYGnV_YtYvmzxzw6A4eEVYpbmi3Y/edit?usp=sharing)
+This plugin has been installed, alone with the [General API Plugin](https://github.com/jenkinsci/checks-api-plugin) on [ci.jenkins.io](https://ci.jenkins.io/Plugins) to help maintain over 1500 Jenkins plugins. You can take a look at the [action](https://github.com/jenkinsci/github-checks-plugin/runs/1025018883) for this repository or other plugin repositories under [Jenkins organization](https://github.com/jenkinsci) for the results.
 
+## Features
+
+*Prerequsite: only GitHub App with proper permissions can publish checks, this [guide](https://github.com/jenkinsci/github-branch-source-plugin/blob/master/docs/github-app.adoc) helps you authenticate your Jenkins as GitHub App.*
```
Comment (timja) [inline README.md:18]:
```suggestion
*Prerequsite: only GitHub App with proper permissions can publish checks, this [guide](https://github.com/jenkinsci/github-branch-source-plugin/blob/master/docs/github-app.adoc) helps you authenticate your Jenkins as a GitHub App.*
```

File: README.md [hunk 3]
```diff
@@ -3,12 +3,41 @@
...
+![GitHub Status](docs/images/github-status.png)
+
+By listening to the Jenkins builds, this plugin will automatically publish statuses (pending, in progress, and completed) to GitHub.
+
+### Rerun Failed Build
+
+![Failed Checks](docs/images/failed-checks.png)
+
+If your Jenkins build failed, a failed check will be published here.
+Then a "Re-run" button will be added automatically by GitHub, by clicking it, you can schedule a new build for the **last** commit of this branch.
```
Comment (timja) [inline README.md:31]:
```suggestion
A "Re-run" button will be added automatically by GitHub, you can schedule a new build for the **last** commit of this branch by clicking the re-run button.
```

File: README.md (hunk 1 above)
Comment (XiongKezhi) [reply README.md:11]: yes... I don't what else needs to write here.

**Review by timja — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -23,6 +23,9 @@ This plugin has been installed, alone with the [General API Plugin](https://gith
 
 By listening to the Jenkins builds, this plugin will automatically publish statuses (pending, in progress, and completed) to GitHub.
 
+*Note: If you are using [GitHub Branch Source Plugin](https://github.com/jenkinsci/github-branch-source-plugin), it will also send status notifications to GitHub through [Status API](https://docs.github.com/en/rest/reference/repos#statuses).
```
Comment (timja) [inline README.md:26]: have you tested these out btw?

Comment (XiongKezhi) [reply README.md:26]: yes, both work on my machine

**Pull request merged**